### PR TITLE
Use keyword arguments for resource classes

### PR DESCRIPTION
**Pull request by 2ykwang**
**Problem**

Use keyword arguments for resource classes

**Solution**

add kwargs to resource class constructor and assign to instance variable.

### CODE REVIEW

**Comment by matthewhegarty** [comment]
Thanks for this but I don't think we need it.  Most of the public api methods take args / kwargs already.  I think that if we pass them into the constructor and then store them it will cause confusion for users, because they won't know whether to use the internal kwargs or those passed into the methods.

Is there some use case you have where it is needed and the current API doesn't solve it?

**Comment by 2ykwang** [comment]
@matthewhegarty 


ex:)

```python
from django.contrib import admin
from polls.models import Poll
from import_export import resources
from import_export.admin import ImportExportModelAdmin

class PoolResource(resources.ModelResource):
    class Meta:
        model = Poll

class PollAdmin(ImportExportModelAdmin):
    resource_classes = [PoolResource]
    def get_export_resource_kwargs(self, request, *args, **kwargs):
        return {“a”:2}

admin.site.register(Poll,PollAdmin)
```

The code above will throw the error `Resource.__init__() got an unexpected keyword argument 'ff'`. because `get_resource_kwargs` method exists, but the constructor cannot handle kwargs.

I think document an example of handling get_resource_kwargs by inheriting the Resource class, or implement code that handles kwargs like the DRF serializer context.

**Comment by matthewhegarty** [comment]
ok - thanks for clarifying.  This looks like issue #1510.  I'll take a look.

File: import_export/resources.py [hunk 1]
```diff
@@ -268,13 +268,14 @@ class Resource(metaclass=DeclarativeMetaclass):
...
 
-    def __init__(self):
+    def __init__(self, *args, **kwargs):
         # The fields class attribute is the *class-wide* definition of
         # fields. Because a particular *instance* of the class might want to
         # alter self.fields, we create self.fields here by copying cls.fields.
         # Instances should always modify self.fields; they should not modify
         # cls.fields.
         self.fields = deepcopy(self.fields)
+        self.kwargs = kwargs
```
Comment (matthewhegarty) [inline import_export/resources.py:278]: It looks like we do need to support kwargs being passed to the constructor (ie. #1517).   However I don't think we should assign them to this internal variable.  Instead I think we should document that the kwargs can be used to pass dynamic parameters to the Resource and it is up to users to handle this in subclasses.

So I suggest we remove this line.

**Review by matthewhegarty — CHANGES_REQUESTED** [review]
Thanks again for raising this.  It is the same as in #1510.  The code has allowed for kwargs to be passed to Resource constructors for a while, but the constructors have never supported this.  I guess it just hasn't come up until now.

I suggest some minor changes to remove the assignment and to document why kwargs is there.  I would also like to see a test which triggers the original issue.  I can help with this if you like?

**Comment by 2ykwang** [comment]
lgtm, thanks for the correction! @matthewhegarty

**Comment by matthewhegarty** [comment]
Feel free to add your name to AUTHORS (if not already).  I'll merge it later.  Thanks again for raising the issue and PR - we appreciate it!

**Comment by 2ykwang** [comment]
I also thank you ! @matthewhegarty 
(FYI I already exist in AUTHORS .

**Pull request merged**